cinema/views.py

```diff
@@ -56,3 +100,67 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset.select_related("movie")
+
+        if self.action == "list":
+            queryset = (
+                queryset
+                .annotate(
+                    tickets_available=(
+                        F("cinema_hall__rows") * F("cinema_hall__seats_in_row")
+                        - Count("tickets")
+                    )
+                )
+            )
+
+            movie = self.request.query_params.get("movie")
+            date = self.request.query_params.get("date")
+
+            if date:
+                date = self._str_to_date(date)
+                queryset = queryset.filter(show_time__date=date)
+            if movie:
+                queryset = queryset.filter(movie__id=int(movie))
+
+        return queryset.distinct()
+
+
+class TicketViewSet(viewsets.ModelViewSet):
+    queryset = Ticket.objects.all()
+    serializer_class = TicketSerializer
+
+
+class OrderSetPagination(PageNumberPagination):
+    page_size = 2
+    page_size_query_param = "page_size"
+    max_page_size = 20
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    pagination_class = OrderSetPagination
+
+    def get_queryset(self):
+        queryset = self.queryset.filter(user=self.request.user)
+
+        if self.action == "list":
+            queryset = (
```

Comment:
This is a bit of a weird hyphenation, I think it would be better to do it like this
```
queryset = queryset.prefetch_related(
...
)
```